### DISCUSSION TITLE
Stores / Volumes

### DISCUSSION THREAD

**Discussion by v-kessler** [lakekeeper/lakekeeper#409] [Ideas]
Allow lakekeeper to add object stores where user can have files that can be used in AI/ML process. 

Lakekeeper should manage governance and metadata for the stores

**Comment by vitaly-am** [comment]
please let me know when this feature will be released , we a really need this

**Reply by v-kessler** [reply]
Can you provide more details on the use case and tech stack. Is it python or something else that will connect to Lakekeeper? you can DM me viktor@vakamo.com, thx

**Comment by giuliotalarico-wr** [comment]
Hi, I'm also very interested in this feature for handling object storage alongside tables for fully integrated lakehouse control from the Lakekeeper UI. Do you have any updates? @v-kessler

**Comment by scaleoutsean** [comment]
This week I explored OpenSharing Volumes and it looks interesting and promising. Use cases are exactly AI/ML, but also traditional batch processing jobs for large files (where latency isn't important), and access by clients that don't understand Tables, and so on. It's as convenient as S3, but better matches existing approaches for centralized access control, logging, security, auditing than S3 access where clients use own S3 credentials.

**Reply by v-kessler** [reply]
@scaleoutsean indeed its interesting. One thing is though its controlled by Detabricks so similar ad Delta hard to build a independent standard I guess. There is something upcoming in Apache Software Foundation around this topic. stay tuned

**Reply by scaleoutsean** [reply]
That's an understandable concern. This month's [announcement](https://www.databricks.com/company/newsroom/press-releases/databricks-announces-opensharing) mentions a lot of stakeholders and says OpenSharing is hosted by the Linux Foundation, so it doesn't look too bad.

My simple notes on OpenSharing Volumes are [here](https://scaleoutsean.github.io/2026/06/17/opensharing-files-versity-s3-gateway-netapp-eseries.html). One thing I noted is that the Volumes [spec](https://github.com/OpenSharing-IO/OpenSharing/blob/main/spec/protocols/VOLUMES.md) seems to have received less effort (it's somewhat superficial) which may be an indicator of newness, but it could also be due to less attention or low participation (outside of Tables spec). On the upside there are few other asset specs being worked on, so I'm curious how non-Tables specs develop - that's one of the things I'll be looking at.
